apps/prairielearn/elements/pl-dropdown/pl-dropdown.py

```diff
@@ -172,17 +176,22 @@ def grade(element_html, data):
 
 def test(element_html, data):
     element = lxml.html.fragment_fromstring(element_html)
+    allow_blank = pl.get_boolean_attrib(element, "allow-blank", ALLOW_BLANK_DEFAULT)
     answers_name = pl.get_string_attrib(element, "answers-name")
     weight = pl.get_integer_attrib(element, "weight", WEIGHT_DEFAULT)
 
     # solution is what the answer should be
     solution = get_solution(element, data)
 
+    result = data["test_type"]
     # incorrect and correct answer test cases
-    if data["test_type"] == "correct":
+    if result == "invalid" and allow_blank:
+        result = "correct"
```

Comment:
I believe you misunderstand the purpose of `test`. This is used to test that the autograder is indeed producing the expected result and works for correct, incorrect and invalid entries. This function creates a random submission that is either correct, incorrect or invalid, depending on what is being tested.

Reply:
You're right, I based my changes on the pl-string-input implementation, but there the special handling in `test` makes sense (because if you allow blanks, it is virtually impossible to have an invalid string input anymore). I've reverted my changes to the function.